test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTypesTest.java

```diff
@@ -0,0 +1,1215 @@
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, {1, 2}, {3, 4}, 10)")
+                .queryColumns("a[1]",
+                              rows(row(1)),
+                              rows(row(0, set(1, 2), set(3, 4), 10)),
+                              rows(row(0, set(1), set(3, 4), null))) // unselected 'b' is repaired, see CASSANDRA-10657
+                .deleteColumn("UPDATE %s SET a=a-{2} WHERE k=0",
+                              "b[4]",
+                              rows(row(4)),
+                              rows(row(0, set(1), set(3, 4), 10)),
+                              rows(row(0, set(1), set(3, 4), 10))) // unselected 'a' is repaired, see CASSANDRA-10657
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, set(1), set(3, 4), 10)))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen lists.
+     */
+    @Test
+    public void testNonFrozenList()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a list<int>, b list<int>, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, [1, 2], [3, 4], 10)")
+                .queryColumns("a",
+                              rows(row(list(1, 2))),
+                              rows(row(0, list(1, 2), list(3, 4), 10)),
+                              rows(flush ? row(0, list(1, 2), list(null, null), null)
+                                         : row(0, list(1, 2), list(3, 4), null))) // TODO: is this RR propagation correct?
+                .deleteColumn("DELETE a[1] FROM %s WHERE k=0",
+                              "b",
+                              rows(row(list(3, 4))),
+                              rows(row(0, list(1), list(3, 4), 10)),
+                              rows(row(0, list(1), list(3, 4), 10))) // unselected 'a' is repaired, see CASSANDRA-10657
+                .deleteRows("DELETE FROM %s WHERE k=0",
+                            rows(),
+                            rows(row(0, list(1), list(3, 4), 10)))
+                .tearDown();
+    }
+
+    /**
+     * Test unrestricted queries with non-frozen maps.
+     */
+    @Test
+    public void testNonFrozenMap()
+    {
+        tester().withQueryRestriction("")
+                .createTable("CREATE TABLE %s (k int PRIMARY KEY, a map<int,int>, b map<int,int>, c int)")
+                .populateNode("INSERT INTO %s (k, a, b, c) VALUES (0, {1:10, 2:20}, {3:30, 4:40}, 10)")
+                .queryColumns("a[2]",
+                              rows(row(20)),
+                              rows(row(0, map(1, 10, 2, 20), map(3, 30, 4, 40), 10)),
+                              rows(flush ? row(0, map(2, 20), map(3, null, 4, null), null)
+                                         : row(0, map(2, 20), map(3, 30, 4, 40), null))) // TODO: is this correct?
```

Comment:
Is this saying that node 2 will contain keys but not values for `b` after the read repair for only `a` but before the read repair on all columns? That...seems like a bug?